chia/wallet/wallet.py

```diff
@@ -332,7 +333,10 @@ async def _generate_unsigned_transaction(
                 raise ValueError(f"Can't send more than {max_send} in a single transaction")
 
         if coins is None:
-            coins = await self.select_coins(total_amount)
+            exclude_coins_list: Optional[List[Coin]] = None
```

Comment:
We'll need a test to exercise these cases.

Reply:
Right. I'd love to add some tests, but as i stated above im a bit lost on where tests start and end, currently they seem to be a little messy. Any pointers are appreciated 👍